### PR TITLE
Move from Travis CI to Github Actions.

### PR DESCRIPTION
**Pull request by litghost**
Additional changes required to make CI green:
 - ae1ae41 Tile type index is now encoded instead of tile type name.

### CODE REVIEW

**Comment by litghost** [comment]
Run is green: https://github.com/litghost/python-fpga-interchange/actions/runs/416525121

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+on: [push, pull_request]
+
+jobs:
+
+  Run-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8']
```
Comment (HackerFoo) [inline .github/workflows/ci.yml:11]: Why not 3.9?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install
+      run: |
+        sudo apt-get update
+        sudo apt-get install clang-format-5.0 xsltproc git
```
Comment (HackerFoo) [inline .github/workflows/ci.yml:26]: Why `clang-format-5.0`?

File: .github/workflows/format.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+make format
+test $(git status --porcelain | wc -l) -eq 0 || { git diff; false;  }
```
Comment (HackerFoo) [inline .github/workflows/format.sh:4]: Why not put this in the workflow? It seems consistent to create a script for these two lines.

File: .github/workflows/test.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+export INTERCHANGE_SCHEMA_PATH="$GITHUB_WORKSPACE/env/RapidWright/interchange"
+export RAPIDWRIGHT_PATH="$GITHUB_WORKSPACE/env/RapidWright"
+
+# Create the device resource for the test part.
+pushd "$GITHUB_WORKSPACE/env" && \
+    "$RAPIDWRIGHT_PATH/scripts/invoke_rapidwright.sh" \
+    com.xilinx.rapidwright.interchange.DeviceResourcesExample \
+    xc7a50tfgg484-1 && popd
+export DEVICE_RESOURCE_PATH="$GITHUB_WORKSPACE/env"
+make test-py
```
Comment (HackerFoo) [inline .github/workflows/test.sh:13]: This is hard to read. Please break up lines 7-11.

Do the environment variables at the top only affect RapidWright, and the one on 12 affect the test? It would be good to comment that.

File: fpga_interchange/device_resources.py [hunk 5]
```diff
@@ -558,7 +558,7 @@ def __init__(self, device_resource_capnp):
             tile_name = self.strs[tile.name]
             tile_name_index = self.string_index[tile_name]
             assert tile_name not in self.tile_name_to_tile
-            tile_type_index = self.tile_type_to_idx[tile.type]
+            tile_type_index = tile.type
```
Comment (HackerFoo) [inline fpga_interchange/device_resources.py:561]: This seems unrelated to the move to GH actions.

File: fpga_interchange/device_resources.py [hunk 6]
```diff
@@ -729,7 +729,7 @@ def get_site_pin(self, site, site_pin_index):
 
         """
         tile = self.device_resource_capnp.tileList[site.tile_index]
-        tile_type_index = self.tile_type_to_idx[tile.type]
+        tile_type_index = tile.type
```
Comment (HackerFoo) [inline fpga_interchange/device_resources.py:732]: And also this.

**Review by HackerFoo — CHANGES_REQUESTED** [review]

File: fpga_interchange/device_resources.py (hunk 5 above)
Comment (litghost) [reply fpga_interchange/device_resources.py:561]: Yes, but it is required because the CI would be red.  This is a result of a change in the upstream schema.

File: fpga_interchange/device_resources.py (hunk 6 above)
Comment (litghost) [reply fpga_interchange/device_resources.py:732]: https://github.com/SymbiFlow/python-fpga-interchange/pull/5#discussion_r541451871

File: .github/workflows/format.sh (hunk 3 above)
Comment (litghost) [reply .github/workflows/format.sh:4]: Do you mean "inconsistent"?  Anyways, I prefer to keep logic out of the github action YAML where possible.  It makes it easier to invoke locally, etc.

File: .github/workflows/test.sh (hunk 4 above)
Comment (litghost) [reply .github/workflows/test.sh:13]: Done

File: .github/workflows/ci.yml (hunk 1 above)
Comment (litghost) [reply .github/workflows/ci.yml:11]: Sure

File: .github/workflows/ci.yml (hunk 2 above)
Comment (litghost) [reply .github/workflows/ci.yml]: This yaml is a copy/paste, so this is an unneeded carry over.

**Comment by litghost** [comment]
Latest CI run: https://github.com/litghost/python-fpga-interchange/actions/runs/416588226

File: fpga_interchange/device_resources.py (hunk 5 above)
Comment (litghost) [reply fpga_interchange/device_resources.py:561]: For clarity, this is in a seperate commit from the other stuff: https://github.com/SymbiFlow/python-fpga-interchange/pull/5/commits/ae1ae41db9082a62bc9750148e04d71802904faa

Comment (HackerFoo) [reply fpga_interchange/device_resources.py:561]: Why the change? What does it fix?

Comment (litghost) [reply fpga_interchange/device_resources.py:561]: Did you read the commit comment?

File: .github/workflows/format.sh (hunk 3 above)
Comment (HackerFoo) [reply .github/workflows/format.sh:4]: Yes, but that is not done in other steps, such as https://github.com/SymbiFlow/python-fpga-interchange/pull/5/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR30-R35.

Personally, I like the idea of putting the logic that could be CI independent in either a Makefile or a single script, like a CI abstraction layer.

File: fpga_interchange/device_resources.py (hunk 5 above)
Comment (HackerFoo) [reply fpga_interchange/device_resources.py:561]: Alright, I'm putting this here in case anyone else looks at this PR:

> Tile type index is now encoded instead of tile type name.

Please update the PR description. These PRs often serve as documentation when I'm trying to understand a change (in addition to Git commit messages.)

**Review by HackerFoo — CHANGES_REQUESTED** [review]

**Review by HackerFoo — APPROVED** [review]
